climada/entity/impact_funcs/trop_cyclone.py

```diff
@@ -311,118 +441,74 @@ def get_countries_per_region(region=None):
-                "DEU", "GHA", "GIB", "GRC", "GRL", "GGY", "GIN", "GNB", "HMD", "VAT",
-                "HUN", "ISL", "IRL", "IMN", "ITA", "JEY", "KEN", "PRK", "XKX", "LVA",
-                "LSO", "LBR", "LBY", "LIE", "LTU", "LUX", "MLT", "MRT", "MYT", "MDA",
-                "MCO", "MNE", "MAR", "NAM", "NLD", "NER", "NGA", "MKD", "NOR", "PSE",
-                "POL", "PRT", "ROU", "RUS", "RWA", "REU", "BLM", "MAF", "SPM", "SMR",
-                "STP", "SEN", "SRB", "SYC", "SLE", "SGP", "SVK", "SVN", "SGS", "SSD",
-                "ESP", "SDN", "SJM", "SWE", "CHE", "TGO", "TUN", "TUR", "UKR", "GBR",
-                "UMI", "ESH", "ZMB", "ALA",
-            ],
-        }
-        # fmt: on
-        impf_id = {
-            "NA1": 1,
-            "NA2": 2,
-            "NI": 3,
-            "OC": 4,
-            "SI": 5,
-            "WP1": 6,
-            "WP2": 7,
-            "WP3": 8,
-            "WP4": 9,
-            "ROW": 10,
-        }
-        region_name = dict()
-        region_name["NA1"] = "Caribbean and Mexico"
-        region_name["NA2"] = "USA and Canada"
-        region_name["NI"] = "North Indian"
-        region_name["OC"] = "Oceania"
-        region_name["SI"] = "South Indian"
-        region_name["WP1"] = "South East Asia"
-        region_name["WP2"] = "Philippines"
-        region_name["WP3"] = "China Mainland"
-        region_name["WP4"] = "North West Pacific"
 
         if region == "all":
-            return region_name, impf_id, iso3n, iso3a
+            return (
+                CountryCode.REGION_NAME.value,
+                CountryCode.IMPF_ID.value,
+                CountryCode.ISO3N.value,
+                CountryCode.ISO3A.value,
+            )
+
+        return (
+            CountryCode.REGION_NAME.value[region],
+            CountryCode.IMPF_ID.value[region],
+            CountryCode.ISO3N.value[region],
+            CountryCode.ISO3A.value[region],
+        )
+
+    @staticmethod
+    def get_impf_id_regions_per_countries(
+        countries: list = None, code_type: str = "ISO3A"
+    ) -> tuple:
+        """Return the impact function id and the region corresponding to a list of countries,
+        or a single country.
+
+        Parameters:
+        -----------
+        countries : list
+            List containing the ISO codes of the country, which should be either
+            in string format if the code is "ISO3A" or an integer if "ISO3N", see code_type below.
+        code_type : str
+            Either "ISO3A" or "ISO3N". "ISO3A" stands for "ISO 3166-1 alpha-3" which is a
+            three-letter country code, "ISO3N" stands for "ISO 3166-1 numeric" which is a
+            three-digit country code, the numeric version of "ISO3A". For example, for Switzerland:
+            the "ISO3A" code is "CHE" and the "ISO3N" is 756.
+
+        Returns:
+        --------
+        impf_ids : list
+            List of impact function ids matching the countries.
```

Comment:
Why not immediately return the appropriate impact function (set)s?

Reply:
Don't you typically define an impact function id column in the exposure ?

Reply:
Agree, I did not saw the other 2 functions. Then I guess we just need to use `CountryCode`in these two.